californium-core/src/main/java/org/eclipse/californium/core/network/deduplication/CropRotation.java

```diff
@@ -48,12 +50,12 @@
 	private boolean running = false;
 	private ScheduledExecutorService executor;
 
-	private ExchangeMap[] maps;
-	private int first;
-	private int second;
+	private final ExchangeMap maps[];
+	private volatile int first;
+	private volatile int second;
```

Comment:
Oh, I see. It is more about the `find` methods' unsynchronized access to the fields, understood.